x-pack/plugins/apm/ftr_e2e/cypress/integration/read_only_user/transaction_detail/transaction_detail.spec.ts

```diff
@@ -0,0 +1,248 @@
+      name: 'breakdownChartRequest',
+    },
+    {
+      method: 'GET',
+      endpoint:
+        '/internal/apm/services/opbeans-java/transactions/traces/samples?*',
+      name: 'getTransactionSamplesRequest',
+    },
+    {
+      method: 'GET',
+      endpoint: '/internal/apm/services/opbeans-java/alerts?*',
+      name: 'getServiceAlertsRequest',
+    },
+    {
+      method: 'GET',
+      endpoint: '/internal/apm/traces/*',
+      name: 'getTraceWaterfallRequest',
+    },
+  ];
+
+  describe('and scrolling down to the trace waterfall', () => {
+    beforeEach(() => {
+      cy.visit(baseUrl);
+      cy.get('[data-test-subj="transactionDetailWaterfallAccordian"]')
+        .first()
+        .as('transactionWaterfallItemWrapper');
+      cy.get('@transactionWaterfallItemWrapper')
+        .find('[data-test-subj="transactionDetailWaterfallAccordian"]')
+        .first()
+        .find('button')
+        .first()
+        .as('spanWaterfallItem');
+      cy.get('@transactionWaterfallItemWrapper')
+        .find('button')
+        .first()
+        .as('transactionWaterfallItem');
+      cy.get('@transactionWaterfallItem').scrollIntoView();
+    });
+
+    it('the services involved in the trace are displayed', () => {
+      cy.get('[data-test-subj="traceWaterfallTimelineLegend"]').as('legend');
+      cy.get('@legend').should('have.length', 2);
+      cy.get('@legend').contains('opbeans-java');
+      cy.get('@legend').contains('postgresql');
+    });
+
+    it('items can be collapsed and expanded', () => {
+      cy.get('[data-test-subj="transactionDetailWaterfallAccordian"]').should(
+        'have.length',
+        2
+      );
+      cy.get('@transactionWaterfallItem').should(
+        'have.attr',
+        'aria-expanded',
+        'true'
+      );
+      cy.get('@transactionWaterfallItem').click('left');
+      cy.get('@transactionWaterfallItem').should(
+        'have.attr',
+        'aria-expanded',
+        'false'
+      );
+      cy.get('@transactionWaterfallItem').click('left');
+      cy.get('@transactionWaterfallItem').should(
+        'have.attr',
+        'aria-expanded',
+        'true'
+      );
+    });
+
+    it('opens a flyout when clicking on a transaction or span', () => {
+      cy.get('@transactionWaterfallItem').click();
+      cy.get('[role="dialog"]').contains('h4', 'Transaction details');
+      cy.get('[data-test-subj="euiFlyoutCloseButton"]').click();
+
+      cy.get('@spanWaterfallItem').click();
+      cy.get('[role="dialog"]').contains('h2', 'Span details');
+      cy.get('[data-test-subj="euiFlyoutCloseButton"]').click();
+    });
+
+    it('displays metadata when clicking on the metadata tab', () => {
+      cy.get('[data-test-subj="transactionDetailsMetadataTab"]').click();
+      cy.get('[data-test-subj="apmMetadataTable"]');
+    });
+
+    it.skip('displays samples from selected range in latency distribution chart', () => {});
```

Comment:
I think this one you can delete, right?

Reply:
yeah i left it here in case anyone knew of a way to click + drag to select a range in cypress?